MEMMODEL_CONSTRAINTS.md

```diff
@@ -103,6 +103,19 @@ may perceive in that statement.)
   for memory used for atomics.  The situation is somewhat better in
   well-typed C++ but reality is complicated.
 
+## Implementation targeting C and C++ or LLVM
+
+* _Targeting C or C++_: it should be possible to implement
+  SharedArrayBuffer using C or C++.  The C/C++11 memory model allows
+  accesses with a number of orderings, and there should be a mapping
+  from SharedArrayBuffer accesses to C/C++11 orderings. Racy
+  SharedArrayBuffer accesses are required to produce defined values,
+  so cannot be implemented using C/C++ non-atomics.
+* _Targeting LLVM_: implementations may also decide to target
+  LLVM, which may provide for more optimization than targeting C/C++
+  and compiling to LLVM. In particular, LLVM allows unordered access,
+  which does not require per-variable sequential consistency.
```

Comment:
I'm not sure this conveys accurate information: users of C/C++ would use the language's `_Atomic`, `std::mutex` and `std::atomic` primitives, these would then compile to ShardArrayBuffer. Non-atomic access in these languages are racy if not synchronized properly, the same will be true with SharedArrayBuffer.

The difference SAB tries to hit is in guaranteeing values observed when races do occur.

I've outlined details of the differences here: https://github.com/tc39/ecmascript_sharedmem/issues/88

LLVM isn't that different from the C++ memory model, except it's on a per-access basis and also supports a Java-oriented memory access (which we should ignore here). I'd also mention other compilers if we're to mention LLVM.

I think it's worth mentioning languages such as Rust which have a pretty different memory model (which under the covers can lower to something like LLV IR).